### PR TITLE
Remove and refactor code deprecated by bootstrap

### LINKED ISSUE #1811: Remove code obsoleted by common bootstrap package
Dependent upon completion of #1801, #1802, #1803, #1804, #1805, #1806, #1807, #1808, #1809, and #1810, and https://github.com/edgexfoundry/edgex-go/issues/2236.

### PR DESCRIPTION
**Pull request by brandonforster**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
This PR refactors the global configuration code to move it into the individual consuming services. It does away with any remaining code that has been deprecated by bootstrap (such as the TOML "testing" code) and attempts to move all code that does not need to be in a common package scope into a local package scope.

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

Issue Number: Fixes #1811

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Are there any specific instructions or things that should be known prior to reviewing?
This code contains no functional changes and can be verified with unit tests and ensuring that services still load and start.

### CODE REVIEW

**Comment by brandonforster** [comment]
I added @tingyuz as a reviewer since I touched security related code with this PR. I did my best to ensure no functional changes, but please verify that assumption.

File: cmd/core-command/toml_test.go [hunk 1]
```diff
@@ -1,25 +0,0 @@
-//
```
Comment (michaelestrin) [inline cmd/core-command/toml_test.go:1]: Why are toml_tests being removed?  Seems like these are valuable in that they ensure the embedded configuration files are valid during the build pipeline.

File: internal/core/data/config/config.go [hunk 2]
```diff
@@ -42,6 +42,30 @@ type WritableInfo struct {
...
+	// Provides additional configuration properties which do not fit within the existing field.
+	// Typically the key is the name of the configuration property and the value is a string representation of the
+	// desired value for the configuration property.
+	Optional map[string]string
+}
+
+// URL constructs a URL from the protocol, host and port and returns that as a string.
+func (m MessageQueueInfo) URL() string {
+	uri := fmt.Sprintf("%s://%s:%v", m.Protocol, m.Host, m.Port)
+	return uri
```
Comment (michaelestrin) [inline internal/core/data/config/config.go:66]: Why not return the result of `fmt.Sprintf()`; `uri` variable is not needed.

File: internal/pkg/bootstrap/interfaces/database.go [hunk 3]
```diff
@@ -14,10 +14,12 @@
 
 package interfaces
 
-import "github.com/edgexfoundry/edgex-go/internal/pkg/config"
+import (
+	bootstrapConfig "github.com/edgexfoundry/go-mod-bootstrap/config"
+)
```
Comment (michaelestrin) [inline internal/pkg/bootstrap/interfaces/database.go:19]: Not idomatic. ;)

File: internal/security/secrets/mocks/custom.go [hunk 4]
```diff
@@ -15,27 +15,27 @@
 package mocks
 
 import (
-	"github.com/edgexfoundry/edgex-go/internal/pkg/config"
+	secretsConfig "github.com/edgexfoundry/edgex-go/internal/security/secrets/config"
 )
```
Comment (michaelestrin) [inline internal/security/secrets/mocks/custom.go:19]: Not idiomatic.  ;)

File: internal/support/scheduler/config/config.go [hunk 5]
```diff
@@ -40,6 +40,49 @@ type WritableInfo struct {
...
+	// Action target API path
+	Path string
+	// Associated Schedule for the Event
+	Interval string
+}
+
+// URI constructs a URI from the protocol, host and port and returns that as a string.
+func (e IntervalActionInfo) URL() string {
+	url := fmt.Sprintf("%s://%s:%v", e.Protocol, e.Host, e.Port)
+	return url
```
Comment (michaelestrin) [inline internal/support/scheduler/config/config.go:83]: `url` is redundant; why not just return result of `fmt.Sprintf()`?

File: internal/system/agent/init.go [hunk 6]
```diff
@@ -128,3 +128,14 @@ func (b *Bootstrap) BootstrapHandler(ctx context.Context, wg *sync.WaitGroup, _
 
 	return true
 }
+
+func listDefaultServices() map[string]string {
```
Comment (michaelestrin) [inline internal/system/agent/init.go:132]: Further limit access by making `listDefaultServices` a method of Bootstrap.

File: cmd/core-command/toml_test.go (hunk 1 above)
Comment (brandonforster) [reply cmd/core-command/toml_test.go:1]: The tests as written leverage code that is marked as deprecated and exists just to run the tests. The value of ensuring that something checked in to source control exists and and is formatted correctly doesn't seem worth having all this extraneous code.

If there's a problem with the TOML, we can identify that in other ways. We don't need a scaffold for that.

**Pull request merged**